File: assignments/password_check/python/password_check_Test_runner.py
import sys
from password_check import PasswordCheck # Import the student's class

def run_tests():
    solver = PasswordCheck()
    # Test Cases: (password, expectedStrength)
    test_cases = [
        # Weak
        ("pass", "Weak"), ("password", "Weak"), ("Password", "Weak"), ("PASSWORD", "Weak"),
        ("12345678", "Weak"), ("!@#$%^&*", "Weak"), ("Pass12", "Weak"), ("LongButOnlyLower", "Weak"),
        ("HASUPPERANDSYMBOL!", "Weak"), (None, "Weak"), ("", "Weak"),
        # Medium
        ("Password123", "Medium"), ("p@sswordLongEnough", "Medium"), ("PASSWORD123$", "Medium"),
        ("12345!@#$LongEnough", "Medium"), ("ThisIsMedium1", "Medium"),
         # Strong
        ("P@ssword123", "Strong"), ("Str0ngP@ss!", "Strong"), ("#GoodP4ssw0rd!", "Strong"),
        ("1A_b2C-d3E=", "Strong")
    ]
    passed_count = 0
    report = []
    print("===== Starting Automated Tests for Password Strength (Python) =====")

    for i, (password, expected) in enumerate(test_cases):
        actual = "Error"
        input_desc = "None" if password is None else f'"{password}"'
        case_identifier = f'Test Case #{i + 1}: Input {input_desc}'
        try:
            actual = solver.check_password(password)
            report.append(case_identifier)
            report.append(f' -> Expected: "{expected}", Got: "{actual}"')

            if actual == expected:
                report.append(" -> ✅ PASS\n")
                passed_count += 1
            else:
                report.append(" -> ❌ FAIL\n")
        except Exception as e:
            report.append(case_identifier)
            report.append(f" -> ❌ ERROR during check: {type(e).__name__}\n")

    print("\n".join(report))
    print("===== Test Summary ====="); print(f"Total Tests: {len(test_cases)}"); print(f"Passed: {passed_count}"); print(f"Failed: {len(test_cases) - passed_count}"); print("========================")
    print(f"[SUMMARY]:{passed_count}:{len(test_cases)}")

if __name__ == "__main__":
    sys.path.append('.')
    run_tests()

File: assignments/password_check/javascript/password_check_test_runner.js
const assert = require('assert');
const { PasswordCheck } = require('./password_check.js'); // Assuming student file

function runTests() {
    const solver = new PasswordCheck();
    // Test Cases: { password: string|null, expected: string }
    const testCases = [
        // Weak
        { password: "pass", expected: "Weak" }, { password: "password", expected: "Weak" },
        { password: "Password", expected: "Weak" }, { password: "PASSWORD", expected: "Weak" },
        { password: "12345678", expected: "Weak" }, { password: "!@#$%^&*", expected: "Weak" },
        { password: "Pass12", expected: "Weak" }, { password: "LongButOnlyLower", expected: "Weak" },
        { password: "HASUPPERANDSYMBOL!", expected: "Weak" }, { password: null, expected: "Weak" },
        { password: "", expected: "Weak" },
        // Medium
        { password: "Password123", expected: "Medium" }, { password: "p@sswordLongEnough", expected: "Medium" },
        { password: "PASSWORD123$", expected: "Medium" }, { password: "12345!@#$LongEnough", expected: "Medium" },
        { password: "ThisIsMedium1", expected: "Medium" },
         // Strong
        { password: "P@ssword123", expected: "Strong" }, { password: "Str0ngP@ss!", expected: "Strong" },
        { password: "#GoodP4ssw0rd!", expected: "Strong" }, { password: "1A_b2C-d3E=", expected: "Strong" }
    ];

    let passedCount = 0;
    let report = [];
    console.log("===== Starting Automated Tests for Password Strength (JavaScript) =====");

    testCases.forEach((tc, i) => {
        let actual = "Error";
        const inputDesc = tc.password === null ? "null" : `"${tc.password}"`;
        const caseIdentifier = `Test Case #${i + 1}: Input ${inputDesc}`;
        try {
            actual = solver.checkPassword(tc.password);
            assert.strictEqual(actual, tc.expected);
            report.push(`${caseIdentifier} -> ✅ PASS (Result: "${actual}")`);
            passedCount++;
        } catch (error) {
             report.push(`${caseIdentifier} -> ❌ FAIL/ERROR: Expected "${tc.expected}", but got "${actual}"`);
        }
        report.push(''); // Add blank line
    });

    console.log(report.join('\n'));
    console.log("===== Test Summary =====");
    console.log(`Total Tests: ${testCases.length}`); console.log(`Passed: ${passedCount}`); console.log(`Failed: ${testCases.length - passedCount}`); console.log("========================");
    console.log(`[SUMMARY]:${passedCount}:${testCases.length}`);
}

runTests();